File: c-sharp/SpaceProbabilities.cs
namespace count_min_sketch
{
    public class SpaceProbabilities
    {
        private const double MILKY_WAY_SIZE = 9500000000000; // 100000 light years

        private const double MILKY_WAY_STARS =  1000000000;

        private const double HUMAN_POPULATION = 9500000000;

        public const double STAR = MILKY_WAY_STARS / MILKY_WAY_SIZE;

        public const double PLANET = 4 * STAR;

        public const double MOON = 2 * PLANET;

        public const double HUMAN = 10 / HUMAN_POPULATION / MILKY_WAY_SIZE;

        public const double PETUNIA_WHALE = 2 * HUMAN / 10;

        public const double HITCHHIKER = 2 * HUMAN / 10;

        public const double SPACE = 1 - (STAR + PLANET + MOON + HUMAN + PETUNIA_WHALE);

        public override string ToString()
        {
            return "Probabilities of space:\n"
                + $"Space: {SPACE:#.##E+0}\n"
                + $"Stars: {STAR:#.##E+0}\n"
                + $"Planets: {PLANET:#.##E+0}\n"
                + $"Moons: {MOON:#.##E+0}\n"
                + $"Humans: {HUMAN:#.##E+0}\n"
                + $"Petunias: {PETUNIA_WHALE:#.##E+0}\n"
                + $"Whales: {PETUNIA_WHALE:#.##E+0}\n"
                + $"Hitchhikers: {HITCHHIKER:#.##E+0}";
        }
    }
}

File: src/info/jackpines/hhgttg/SpaceProbabilities.java
package info.jackpines.hhgttg;

public class SpaceProbabilities {
  private static double MILKY_WAY_SIZE = 9_500_000_000_000.0; // 100000 light years

  private static double MILKY_WAY_STARS = 1_000_000_000.0;

  private static double HUMAN_POPULATION = 9_500_000_000.0;

  public static double STAR = MILKY_WAY_STARS / MILKY_WAY_SIZE;

  public static double PLANET = 4 * STAR;

  public static double MOON = 2 * PLANET;

  public static double HUMAN = 10 / HUMAN_POPULATION / MILKY_WAY_SIZE;

  public static double PETUNIA_WHALE = 2 * HUMAN / 10;

  public static double HITCHHIKER = 2 * HUMAN / 10;

  public static double SPACE = 1 - (STAR + PLANET + MOON + HUMAN + PETUNIA_WHALE);

  @Override
  public String toString()
  {
    return String.format( "Probabilities of space:%n"
            + "Space: %e%n"
            + "Stars: %e%n"
            + "Planets: %e%n"
            + "Moons: %e%n"
            + "Humans: %e%n"
            + "Petunias: %e%n"
            + "Whales: %e%n"
            + "Hitchhikers: %e",
        SPACE, STAR, PLANET, MOON, HUMAN, PETUNIA_WHALE, PETUNIA_WHALE, HITCHHIKER );
  }
}
